File: Solucion_Codigo/ProyectoBim1_ SistemaDeclaracionImpuestos/src/Controlador.java
import java.util.Random;

public class Controlador {

    private static final int CANTIDAD_CONTRIBUYENTES = 3;
    private Vista vista = new Vista();

    public void ejecutar() {
        Metodos[] declaraciones = new Metodos[CANTIDAD_CONTRIBUYENTES];

        for (int i = 0; i < CANTIDAD_CONTRIBUYENTES; i++) {
            String nombre = "Contribuyente" + (i + 1);
            String cedula = generarCedulaAleatoria();
            String direccion = "Direccion #" + (100 + i);

            Metodos declaracion = new Metodos(nombre, cedula, direccion);

            for (int mes = 0; mes < 12; mes++) {
                double sueldo = 1000 + Math.random() * 2000;
                declaracion.agregarSueldo(mes, sueldo);
            }

            for (String categoria : Metodos.CATEGORIAS) {
                double montoFactura = Math.random() * 1500;
                declaracion.agregarFactura(montoFactura, categoria);
            }

            declaracion.calcularTotalIngresos();
            declaracion.calcularDeducciones();
            declaracion.calcularImpuestos();

            declaraciones[i] = declaracion;
            vista.mostrarDeclaracion(declaracion);
        }
    }

    private String generarCedulaAleatoria() {
        Random rand = new Random();
        StringBuilder cedula = new StringBuilder();
        for (int i = 0; i < 10; i++) {
            cedula.append(rand.nextInt(10));
        }
        return cedula.toString();
    }
}

File: Solucion_Codigo/Proyecto Bimestral Python/controlador/controlador.py
import random
from modelo.metodos import Metodos
from vista.vista import Vista

class Controlador:
    def ejecutar(self):
        vista = Vista()
        contribuyentes = []

        for i in range(3):
            nombre = f"Contribuyente{i + 1}"
            cedula = ''.join(str(random.randint(0, 9)) for _ in range(10))
            direccion = f"Dirección #{100 + i}"
            persona = Metodos(nombre, cedula, direccion)

            for mes in range(12):
                persona.agregar_sueldo(mes, random.uniform(1000, 3000))

            for categoria in Metodos.CATEGORIAS:
                persona.agregar_factura(random.uniform(0, 1500), categoria)

            persona.calcular_total_ingresos()
            persona.calcular_deducciones()
            persona.calcular_impuestos()

            contribuyentes.append(persona)
            vista.mostrar_declaracion(persona)